relay.go

```diff
@@ -259,6 +259,7 @@ func (r *Relayer) getDestination(f lazyCallReq, cs relay.CallStats) (*Connection
 
 	// Get the destination
 	selectedPeer, err := r.hosts.Get(f)
+	cs.SetPeer(selectedPeer)
```

Comment:
Yep but it might have the pool set so we need to cash this even if there's an error.

If the routing table picks a pool, but then can't find a task within that pool, it will return an error but the peer will contain a zone/pool. We want stats tied to the pool, so we have to set them even if there's an error.

This interface needs work (maybe multiple returns), but we need to figure out the interface for peer management first.